include/gsl/gsl

```diff
@@ -19,32 +19,28 @@
 #ifndef GSL_GSL_H
 #define GSL_GSL_H
 
-#include "gsl_assert"  // Ensures/Expects
-#include "gsl_util"    // finally()/narrow()/narrow_cast()...
-#include "multi_span"  // multi_span, strided_span...
-#include "span"        // span
-#include "string_span" // zstring, string_span, zstring_builder...
+#include <gsl/gsl_assert>  // Ensures/Expects
+#include <gsl/gsl_util>    // finally()/narrow()/narrow_cast()...
+#include <gsl/multi_span>  // multi_span, strided_span...
+#include <gsl/span>        // span
+#include <gsl/string_span> // zstring, string_span, zstring_builder...
 #include <memory>
 
-#ifdef _MSC_VER
-
-// No MSVC does constexpr fully yet
-#pragma push_macro("constexpr")
-#define constexpr /*constexpr*/
-
-// MSVC 2013 workarounds
-#if _MSC_VER <= 1800
-// noexcept is not understood
-#pragma push_macro("noexcept")
-#define noexcept /*noexcept*/
+#if defined(_MSC_VER) && _MSC_VER < 1910
+ #pragma push_macro("constexpr")
+ #define constexpr inline
 
```

Comment:
There's a trade-off being made here. We can either:

* Define `constexpr` as whitespace/empty, and use the abominable `constexpr inline` on functions and `constexpr const` on data,

* Define `constexpr` as `inline` so it can be used directly on functions. Define a separate macro for data (I used `GSL_CONSTEXPR_DATA` in `<gsl/multi_span>`) as either `constexpr` or `const`.

Both options require some non-idiomatic "garbage" syntax. For the first option, *every* usage of `constexpr` must instead be `constexpr inline` (function) or `constexpr const` (data). For the second, `constexpr` functions are written idiomatically, and only `constexpr` data - which is at least two orders of magnitude less common than `constexpr` functions in the current GSL - needs to instead be tagged with `GSL_CONSTEXPR_DATA`.

I prefer option two since it is less intrusive overall. If you disagree, say so and I'll retag all the `constexpr` entities.